### PR TITLE
Scrapper, Sonia Zakharova - 21FPL2

### PR DESCRIPTION
**Pull request by solonarr**

### CODE REVIEW

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
Not seeing any activity in lab_5_scrapper folder. It is right time to start implementing lab 5.
Please, delete all files .html, ьфшт.py and others that are not related to laboratory work assignment.

**Comment by solonarr** [comment]
:'(

**Comment by solonarr** [comment]
> Not seeing any activity in lab_5_scrapper folder. It is right time to start implementing lab 5. Please, delete all files .html, ьфшт.py and others that are not related to laboratory work assignment.

fixed

**Comment by solonarr** [comment]
:'''(

File: lab_5_scrapper/scrapper.py [hunk 1]
```diff
@@ -1,82 +1,149 @@
...
         """
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            config = json.load(f)
+        seed_urls = config['seed_urls']
+        headers = config['headers']
+        total_articles_to_find_and_parse = config['total_articles_to_find_and_parse']
+        encoding = config['encoding']
+        timeout = config['timeout']
+        verify_certificate = config['should_verify_certificate']
+        headless_mode = config['headless_mode']
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: where is validation?

File: lab_5_scrapper/scrapper.py [hunk 2]
```diff
@@ -1,82 +1,149 @@
...
+    headless_mode: bool
 
     def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._validate_config_content()
+        configDTO = self._extract_config_content()
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: please refer to python variables naming style and fix this accordingly

File: lab_5_scrapper/scrapper.py [hunk 3]
```diff
@@ -1,82 +1,149 @@
...
+        headless_mode = config['headless_mode']
+
         pass
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        return configDTO.seed_urls
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: you have specific class attributes for all getting methods -> there is no need to get them from configDTO -> get them from class fields directly e.g. here it is _seed_urls

File: lab_5_scrapper/scrapper.py [hunk 4]
```diff
@@ -90,25 +157,35 @@ def __init__(self, config: Config) -> None:
...
+        self.urls = []
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
+
+        link = article_bs.get('href')
+        if (link is not None) and
         pass
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: pass should be removed in methods that you implement

File: lab_5_scrapper/scrapper.py [hunk 5]
```diff
@@ -90,25 +157,35 @@ def __init__(self, config: Config) -> None:
...
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
+        for link_bs in self.config._extract_config_content().get_seed_urls:
+            link = link_bs.get('href')
+            if link is None:  # no ==
+                continue
+            elif link[0] == 'n' and link[1] == 'e' and link[2] == 'w' and link.count('#') == 0 and link.count('?') == 0:
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: can you suggest other ways of getting links? maybe use regular expressions or check specific string is included or not

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
Good start! Please, fix linting and finalise implementation

File: lab_5_scrapper/scrapper.py (hunk 3 above)
Comment (solonarr) [reply lab_5_scrapper/scrapper.py]: fixed

File: lab_5_scrapper/scrapper.py (hunk 4 above)
Comment (solonarr) [reply lab_5_scrapper/scrapper.py]: fixed

File: lab_5_scrapper/scrapper.py (hunk 1 above)
Comment (solonarr) [reply lab_5_scrapper/scrapper.py]: fixed

File: lab_5_scrapper/scrapper.py (hunk 2 above)
Comment (solonarr) [reply lab_5_scrapper/scrapper.py]: fixed

File: lab_5_scrapper/scrapper.py (hunk 5 above)
Comment (solonarr) [reply lab_5_scrapper/scrapper.py]: fixed

**Comment by vitazaytseva** [comment]
Well done! You can proceed to work on Lab 6)

**Pull request closed without merging**